server/server.go

```diff
@@ -32,55 +62,170 @@ func StartHttpServer(opts *options.Options) error {
+		timeout = time.Second * time.Duration(c.Timeout)
+	}
 
-	conn, err := net.DialTimeout("tcp", fmt.Sprintf("0.0.0.0:%d", port), defaultTimeout)
+	conn, err := net.DialTimeout("tcp", fmt.Sprintf("%s:%d", c.Host, c.Port), timeout)
 	if err != nil {
-		return err
+		return &CheckFail{name: c.Name, reason: err.Error()}
 	}
 
 	defer conn.Close()
+	logger.Infof("Check SUCCESS: %s", c.Name)
+	return nil
+}
 
+func (c HttpCheck) ValidateCheck () error {
+	if c.Name == "" {
+		return &InvalidCheck{name: "http", key: "name"}
+	}
+	if c.Host == "" {
+		return &InvalidCheck{name: c.Name, key: "host"}
+	}
+	if c.Port == 0 {
+		return &InvalidCheck{name: c.Name, key: "port"}
+	}
+	return nil
+}
+
+func (c HttpCheck) DoCheck (opts *options.Options) error {
+	logger := opts.Logger
+	logger.Infof("Checking %s at %s:%d via HTTP...", c.Name, c.Host, c.Port)
+
+	timeout := time.Second * DEFAULT_CHECK_TIMEOUT
+	if c.Timeout != 0 {
+		// override default with user defined timeout
+		timeout = time.Second * time.Duration(c.Timeout)
+	}
+
+	client := http.Client{
+		Timeout: timeout,
+	}
+	resp, err := client.Get(fmt.Sprintf("http://%s:%d", c.Host, c.Port))
+	if err != nil {
+		return &CheckFail{name: c.Name, reason: err.Error()}
+	}
+
+	switch {
+	case len(c.SuccessStatusCodes) > 0:
+		if !contains(c.SuccessStatusCodes, resp.StatusCode) {
+			return &CheckFail{name: c.Name, reason: fmt.Sprintf("wanted one of %v, got %d", c.SuccessStatusCodes, resp.Status)}
+		}
+	case c.BodyRegex != "":
+		defer resp.Body.Close()
+		body, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return &CheckFail{name: c.Name, reason: "failed reading body"}
+		}
+
+		if !strings.Contains(string(body), c.BodyRegex) {
+			return &CheckFail{name: c.Name, reason: fmt.Sprintf("wanted %s in http body, got %s", c.BodyRegex, body)}
+		}
+	default:
+		// no success_codes or body_regex defined, only pass on 200
+		if resp.StatusCode != http.StatusOK {
+			return &CheckFail{name: c.Name, reason: fmt.Sprintf("wanted status code 200, got %d", resp.StatusCode)}
+		}
+	}
+
+	logger.Infof("Check SUCCESS: %s", c.Name)
+	return nil
+}
+
+// TODO: move into helpers
+func contains(s []int, e int) bool {
+	for _, a := range s {
+		if a == e {
+			return true
+		}
+	}
+	return false
+}
+
+func (c ScriptCheck) ValidateCheck () error {
+	if c.Name == "" {
+		return &InvalidCheck{name: "script", key: "name"}
+	}
+	if c.Script == "" {
+		return &InvalidCheck{name: c.Name, key: "script"}
+	}
+	return nil
+}
+
+func (c ScriptCheck) DoCheck (opts *options.Options) error {
+	logger := opts.Logger
+	logger.Infof("Checking %s at %s...", c.Name, c.Script)
+
+	timeout := time.Second * DEFAULT_CHECK_TIMEOUT
+	if c.Timeout != 0 {
+		// override default with user defined timeout
+		timeout = time.Second * time.Duration(c.Timeout)
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	cmd := exec.CommandContext(ctx, c.Script)
+	_, err := cmd.Output()
```

Comment:
Don't throw away the output from the command, as it's essential for debugging what happened. Instead, log that output.